src/components/site/Practice.jsx

```diff
@@ -19,79 +22,111 @@ class Practice extends React.PureComponent {
-        const point = data.galaxies[i];
-        const coor = [];
-        coor.push(point.RA);
-        coor.push(point.Dec);
-        dataArr.push(coor);
-      }
-      this.setState(prevState => ({
-        ...prevState,
-        graphData: dataArr,
-      }));
-    }
-  }
-
-  onChartClick(params){
-    console.log(params);
-    console.log(params.color)
+  onChartClick(params) {
+    this.setState(prevState => ({
+      ...prevState,
+      clickedPt: params.value,
+    }));
   }
 
   getEvent() {
     const onEvents = {
-        'click': this.onChartClick,
-    }
+      click: this.onChartClick,
+    };
     return onEvents;
-  };
+  }
 
   getOption() {
-    const { graphData } = this.state;
-    const obj = {
-      title: {
-        text: 'Exploring Observable Universe',
-      },
-      tooltip: {
-        formatter: params => {
-          const data = params.data || [0, 0];
-          return data[0].toFixed(2) + ', ' + data[1].toFixed(2);
+    const { data, clickedPt } = this.state;
+    let obj = {};
+    if (Object.keys(data).length !== 0) {
+      obj = {
+        animation: false,
+        title: {
+          text: 'Exploring Observable Universe',
         },
-        axisPointer: {
-          type: 'cross',
+        legend: {
+          orient: 'vertical',
+          right: 5,
         },
-      },
-      xAxis: {
-        name: 'RA',
-        nameLocation: 'middle',
-        nameGap: 25,
-      },
-      yAxis: {
-        name: 'Dec',
-        nameLocation: 'middle',
-        nameGap: 25,
-      },
-      series: [
-        {
-          symbolSize: 3,
-          data: graphData,
-          type: 'scatter',
+        dataset: {
+          source: data.galaxies,
+          dimensions: ['RA', 'Dec'],
         },
-      ],
-    };
+        tooltip: {
+          formatter: params => {
+            const dat = params.value;
+            return dat.RA.toFixed(2) + ', ' + dat.Dec.toFixed(2);
+          },
+        },
+        xAxis: {
+          name: 'RA',
+          nameLocation: 'middle',
+          nameGap: 25,
+        },
+        yAxis: {
+          name: 'Dec',
+          nameLocation: 'middle',
+          nameGap: 25,
+        },
+        series: [
+          {
+            name: 'Data Point',
+            symbolSize: 8,
+            type: 'scatter',
+            itemStyle: {
+              normal: {
+                color: params => {
+                  const { value } = params;
+                  const defaultColor = '#e97a7a';
+                  if (value == null) {
+                    return '';
+                  } else if (
+                    clickedPt.RA === value.RA &&
+                    clickedPt.Dec === value.Dec
+                  ) {
+                    return '#b7b7ff';
+                  } else {
+                    return defaultColor;
+                  }
+                },
+              },
+            },
+          },
+        ],
+        dataZoom: [
```

Comment:
I think, if possible, constrain how far one can zoom in even more than you have here.  I can go so far in there that I don't even really know what's going on.